pkg/election/leadership.go

```diff
@@ -376,6 +391,13 @@ func (ls *Leadership) Watch(serverCtx context.Context, revision int64) {
 						zap.Int64("revision", wresp.Header.Revision), zap.String("leader-key", ls.leaderKey), zap.String("purpose", ls.purpose))
 					return
 				}
+				// ONLY `{service}/primary/transfer` API update primary will meet this condition.
+				if ev.Type == mvccpb.PUT && ls.IsPrimary() {
```

Comment:
I think here we can optimize the case that if the updated primary is still itself, no need to return to campaign again.

Reply:
https://github.com/HuSharp/pd/blob/43830ec9ea80d92008be663ec5c26b3d137373a9/pkg/mcs/utils/expected_primary.go#L141-L146

I skipped the member on the outer layer(in `transfer primary`) which is not the same as oldPrimary, do you think I still need to add the restriction inside the `watch`?

Maybe the `expected primary flag` should not be modified when the leader is itself, because this flag needs to keepalive lease after the campaign leader, which means that it requires the Primary to quit the current watch.